client/app/src/main/java/com/example/criengine/FilterFragmentActivity.java

```diff
@@ -0,0 +1,125 @@
+package com.example.criengine;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.CheckBox;
+import android.widget.CompoundButton;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+import java.io.Serializable;
+import java.util.ArrayList;
+
+/**
+ * Contains the fragment that appears if a user wants to filter information when viewing books they
+ * own.
+ */
+public class FilterFragmentActivity extends DialogFragment implements Serializable {
+    private CheckBox avaliableButton;
+    private CheckBox requestsButton;
+    private CheckBox acceptedButton;
+    private ArrayList<String> filterStatus;
+    private OnFragmentInteractionListener listener;
+
+    // Contains the method that the MyBooksActivity needs to implement.
+    public interface OnFragmentInteractionListener {
+        void onConfirmPressed(ArrayList<String> filterStatus);
+    }
+
+    // Constructor.
+    public FilterFragmentActivity() {}
+
+    public FilterFragmentActivity(ArrayList<String> filterStatus) {
+        this.filterStatus = filterStatus;
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (context instanceof OnFragmentInteractionListener){
+            listener = (OnFragmentInteractionListener) context;
+        } else {
+            throw new RuntimeException(context.toString()
+                    + " must implement OnFragmentInteractionListener");
+        }
+    }
+
+    /**
+     * Creates the fragment.
+     * @param savedInstanceState
+     * @return The ArrayList containing the filter status information.
+     */
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        View view = LayoutInflater.from(getActivity()).inflate(R.layout.activity_filter_fragment, null);
+        avaliableButton = view.findViewById(R.id.checkbox_available_filter);
+        requestsButton = view.findViewById(R.id.checkbox_requests_filter);
+        acceptedButton = view.findViewById(R.id.checkbox_accepted_filter);
+        AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+
+        builder
+                .setView(view)
+                .setTitle("Filter By Status")
+                .setNegativeButton("Cancel", null);
+
+        // Get the saved status of the checkbox's.
+        if (filterStatus.contains("Available")) {
+            avaliableButton.setChecked(true);
+        }
+        if (filterStatus.contains("Requests")) {
+            requestsButton.setChecked(true);
+        }
+        if (filterStatus.contains("Accepted")) {
+            acceptedButton.setChecked(true);
+        }
+
```

Comment:
Can probably use a switch case here to remove all the redundant code (kinda a smell) but that's a refactor

Reply:
Seeing your code [here](https://github.com/CMPUT301F20T05/CriEngine/pull/55/files#diff-5beca891d744a8ed71d11cf9d327d96d9bd31be542363e38239fa07cc993fbf9R94), it seems that it is intended to be able to filter to more than one status, with filters associated by OR. If we were to use a switch case, we can't use `break;` since that would be different from the code here. Additionally, I think switch cases are more used for scenarios where a value is any **one** of the cases, so it's not as appropriate to use here where the value can be one or more.

Reply:
this is mostly for the code below, can make a general "CompountButton.OnCheckedChangeListener()" that uses inputs to determine what the function is or a switch that works for all the buttons with different cases depending on the button

Reply:
I'm not really seeing the purpose of the switch statement here. As Cynthia said, we would just end up having to remove the `break` because this filter is intended to be able to select multiple filters at once. At that point I feel like it's not making use of the switch statement anymore and would make the code harder to read (?)

If this was a `if-else` code block then I would see a need for the switch, but not as it is rn

Reply:
it's for the code below the comment, not above all the onclick handlers can be condensed into a general one

Reply:
We can't really see code that's below the comment lines. Would you point them out for us please?

Reply:
yeno the comment shows up in the files changed, didn't realize this is how github works with this kinda thing

Reply:
hes referring to L83-114